alea/test/result.cpp

```diff
@@ -0,0 +1,94 @@
+#include <alps/alea.hpp>
+#include <alps/testing/near.hpp>
+
+#include "gtest/gtest.h"
+
+template <typename Acc>
+class result_case
+    : public ::testing::Test
+{
+public:
+    typedef Acc acc_type;
+    typedef alps::alea::traits<Acc> acc_traits;
+    typedef typename acc_traits::store_type  store_type;
+    typedef typename acc_traits::value_type value_type;
+
+    typedef typename acc_traits::result_type result_type;
+    typedef alps::alea::traits<result_type> result_traits;
+
+    result_case()
+    {
+        acc_type acc(2);
+        for (size_t i = 0; i != 100; ++i)
+            acc << std::array<value_type, 2>{{1.0 * i, 2.0 * i}};
+
+        res = acc.result();
+        generic_res = acc.result();
+    }
+
+    void test_result_caps()
+    {
+        EXPECT_EQ(res.size(), generic_res.size());
+        EXPECT_EQ(res.count(), generic_res.count());
+        ALPS_EXPECT_NEAR(res.mean(), generic_res.mean<value_type>(), 1e-16);
+    }
+
+    // Thank you, C++, for this very nice way of doing "if/else".  Go die.
```

Comment:
Too bad, we cannot have `constexpr if` just yet.
Here is a little bit less annoying way of doing this.
```c++
    // Have VAR
    void test_var(std::true_type) {
        typedef typename result_traits::value_type value_type;
        ALPS_EXPECT_NEAR(res.var(), generic_res.var<value_type>(), 1e-16);
    }

    // Does not have VAR
    void test_var(std::false_type) {
        typedef typename result_traits::value_type value_type;
        EXPECT_THROW(generic_res.var<value_type>(), alps::alea::estimate_unavailable);
    }

...

TYPED_TEST(result_case, test_var) {
    this->test_var(std::integral_constant<bool, TestFixture::result_traits::HAVE_VAR>());
}
```